cypress/integration/deal-groups.js

```diff
@@ -0,0 +1,257 @@
+      cy.route('/api/v1/km-proxy/advertisers?limit=5000&is_active=true&sort=name').as('advertiserList');
+      cy.route('/api/v1/km-proxy/agency-trade-desks?limit=5000&sort=name').as('atd-api');
+      cy.route('/api/v1/users/session-user').as('sessionUser');
+      cy.route('/api/v1/role').as('role');
+
+      cy.visit('');
+      cy.get('[data-qa="deal-group-dashboard--create"]', { timeout: 9000 }).click();
+      cy.url().should('include', 'deal-dashboard/deal-groups/create');
+
+      // Adding basic Deal-Group info
+      cy.get('[data-qa="deal-group-form--name"]').focus().type(dealGroupName);
+      cy.get('[data-qa="deal-group-form--sf-id"]').focus().type(salesforceId);
+      cy.get('[data-qa="deal-group-form--target-spend"]').focus().type(targetSpend);
+      cy.get('[data-qa="deal-group-form--closed-loop"]').click();
+      cy.get('[data-qa="deal-group-form--closed-loop--1"]').click()
+        .wait('@teamMembers')
+        .wait('@advertiserList')
+        .wait('@atd-api')
+        .wait('@role')
+        .wait('@sessionUser'); // making sure these APIs are cleared
+
+
+      // Buyers Section
+      cy.get('[data-qa="deal-group-form--select-bidder"]').click().focus().type('A');
+      cy.get('[data-qa="deal-group-form--select-bidder"]').parent('.input-wrapper--typeahead').find('.dropdown-menu').children('li')
+        .first()
+        .children('a')
+        .click();
+      cy.get('[data-qa="deal-form--select-advertiser"]').click().type('max effort');
+      cy.contains('max effort').click(); // clicking on advertiser dropdown
+
+      // // Team Member Section
+      cy.get('[data-qa="deal-group-form--select--agency_sales_lead"]').click();
+      cy.get('ul > li.active', { timeout: 9000 }).click(); // clicking on 1st option from dropdown
+      cy.get('[data-qa="deal-group-form--select--client_service_manager"]').click();
+      cy.get('ul > li.active', { timeout: 9000 }).click(); // clicking on 1st option from dropdown
+      cy.get('[data-qa="deal-group-form--select--campaign_manager"]').click();
+      cy.get('ul > li.active', { timeout: 9000 }).click(); // clicking on 1st option from dropdown
+
+      // KPI Section
+      cy.get('[data-qa="deal-group-form--optimizable-kpi"]').focus().click();
+      cy.get('[data-qa="deal-group-form--optimizable-kpi--0"]').click();
+      cy.get('[data-qa="deal-group-form--outcome-kpi"]').click();
+      cy.get('[data-qa="deal-group-form--outcome-kpi--0"]').click();
+
+      // Submitting Deal Group info
+      cy.get('[data-qa="deal-group-add-edit--submit"]').click();
+      cy.get('[data-qa="modal--confirm"]').click();
+      cy.get('[data-qa="deal-group-detail--title"]', { timeout: 5000 }).should('contain', dealGroupName);
+
+      cy.location().then((currentLocation) => {
+        const urlPathName = currentLocation.pathname;
+        dealGroupID = urlPathName.split('/').pop(); // Grabbing Deal-Group ID from URL
+      });
+    });
+
+    it('Retrieve the deal-group', () => {
```

Comment:
This should be like: Retrieve deal-group info from api.

However, you should be pulling these values from the UI using the .invoke() method unless you had trouble